assets/js/blocks/checkout/inner-blocks/checkout-collection-method/block.tsx

```diff
@@ -0,0 +1,214 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useShippingData } from '@woocommerce/base-context/hooks';
+import { getCurrencyFromPriceResponse } from '@woocommerce/price-format';
+import FormattedMonetaryAmount from '@woocommerce/base-components/formatted-monetary-amount';
+import {
+	__experimentalRadio as Radio,
+	__experimentalRadioGroup as RadioGroup,
+} from 'wordpress-components';
+import classnames from 'classnames';
+import { Icon, store, shipping } from '@wordpress/icons';
+import { getSetting } from '@woocommerce/settings';
+import type { CartShippingPackageShippingRate } from '@woocommerce/type-defs/cart';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+/**
+ * Returns the cheapest rate that isn't a local pickup.
+ *
+ * @param {Array} shippingRates Array of shipping Rate.
+ *
+ * @return {Object} cheapest rate.
+ */
+function getShippingStartingPrice(
```

Comment:
This is cool but noting it will break if there is no address. Should we let the user set the default text for the from price, or do we need a way of grabbing all shipping rates from all zones for this? If so, might be better to pass the prices to the app from PHP side rather than calculate it.

Reply:
This is a bit tricky here because we want to hide the "Delivery" box if we don't have any shipping rates at all, we can't know for sure unless we calculated shipping right? I think I can use that flag.

Reply:
Well, since that is chicken and egg, perhaps we show the selection regardless and show the no rates found message as we do now, later on. So basically, show this section as long as there are shipping methods activated, not necessarily available rates.